### ISSUE TITLE
[LINT_BUG]: Unexpected interaction between box_unused_att_pkg_fun_linter and nolint

### ISSUE THREAD

**Issue by calderonsamuel** [Appsilon/box.linters#182]
### box.linters version

box.linters: ‘0.10.5’
R 4.5.1
Windows 11

### Sample source code to lint

There is a weird interaction with "nolint" and `box_unused_att_pkg_fun_linter`. Basically, it creates an incorrect lint. In `code2` you can see that `setClass` is being used but not being recognized. 
``` r
code2 <- "box::use(
  methods[
    `slot<-`, # nolint: box_unused_att_pkg_fun_linter
    setClass,
  ],
)

setClass('Person', slots = c(name = 'character'))
person <- new('Person')
slot(person, 'name') <- 'John'
"

lintr::lint(code2, linters = box.linters::box_unused_att_pkg_fun_linter())
#> <text>:4:5: warning: [box.linters::box_unused_att_pkg_fun_linter] Imported function unused.
#>     setClass,
#>     ^~~~~~~~
```

For the previous code, I would expect no lint results. The linter correctly behaves for `code3`, where there is no use of "nolint".

```r
code3 <- "box::use(
  methods[
    setClass,
  ],
)

setClass('Person', slots = c(name = 'character'))
"

lintr::lint(code3, linters = box.linters::box_unused_att_pkg_fun_linter())
#> ℹ No lints found.
```

<sup>Created on 2025-10-21 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### Lint command used

Already provided

### Lint result

Already provided

### Expected result

Already provided
